Tutorials/src/eventMixing.cxx

```diff
@@ -32,112 +35,223 @@ DECLARE_SOA_TABLE(Hashes, "AOD", "HASH", hash::Bin);
-//       int bin = binningOnPositions.getBin({c1.posX(), c1.posY()};
-//       LOGF(info, "Mixed event bin: %d collisions: (%d, %d), tracks: (%d, %d)", bin, c1.globalIndex(), c2.globalIndex(), tracks1.size(), tracks2.size());
-//       for (auto& [t1, t2] : combinations(CombinationsFullIndexPolicy(tracks1, tracks2))) {
-//         LOGF(info, "Mixed event tracks pair: (%d, %d) from events (%d, %d), track event: (%d, %d)", t1.index(), t2.index(), c1.index(), c2.index(), t1.collision().index(), t2.collision().index());
-//       }
-//     }
-//   }
-// };
-//
-// struct MixedEventsInsideProcess {
-//   std::vector<double> xBins{VARIABLE_WIDTH, -0.064, -0.062, -0.060, 0.066, 0.068, 0.070, 0.072};
-//   std::vector<double> yBins{VARIABLE_WIDTH, -0.320, -0.301, -0.300, 0.330, 0.340, 0.350, 0.360};
-//   BinningPolicy<aod::collision::PosX, aod::collision::PosY> binningOnPositions{{xBins, yBins}, true}; // true is for 'ignore overflows' (true by default)
-//
-//   void process(aod::Collisions& collisions, aod::Tracks& tracks)
-//   {
-//     LOGF(info, "Input data Collisions %d, Tracks %d ", collisions.size(), tracks.size());
-//
-//     auto tracksTuple = std::make_tuple(tracks);
-//     SameKindPair<aod::Collisions, aod::Tracks> pair{binningOnPositions, 5, -1, collisions, tracksTuple};
-//
-//     for (auto& [c1, tracks1, c2, tracks2] : pair) {
-//       int bin = binningOnPositions.getBin({c1.posX(), c1.posY()});
-//       LOGF(info, "Mixed event bin: %d collisions: (%d, %d), tracks: (%d, %d)", bin, c1.globalIndex(), c2.globalIndex(), tracks1.size(), tracks2.size());
-//       for (auto& [t1, t2] : combinations(CombinationsFullIndexPolicy(tracks1, tracks2))) {
-//         LOGF(info, "Mixed event tracks pair: (%d, %d) from events (%d, %d)", t1.globalIndex(), t2.globalIndex(), c1.globalIndex(), c2.globalIndex());
-//       }
-//     }
-//   }
-// };
+struct MixedEventsTriple {
+  std::vector<double> xBins{VARIABLE_WIDTH, -0.064, -0.062, -0.060, 0.066, 0.068, 0.070, 0.072};
+  std::vector<double> yBins{VARIABLE_WIDTH, -0.320, -0.301, -0.300, 0.330, 0.340, 0.350, 0.360};
+  std::vector<double> zBins{7, -7, 7};
+  using BinningType = BinningPolicy<aod::collision::PosX, aod::collision::PosY, aod::collision::PosZ>;
+  BinningType binningOnPositions{{xBins, yBins, zBins}, true}; // true is for 'ignore overflows' (true by default)
+  SameKindTriple<aod::Collisions, aod::Tracks, BinningType> triple{binningOnPositions, 5, -1};
+
+  void process(aod::Collisions const& collisions, aod::Tracks const& tracks)
+  {
+    LOGF(info, "Input data Collisions %d, Tracks %d", collisions.size(), tracks.size());
+
+    int count = 0;
+    for (auto& [c1, tracks1, c2, tracks2, c3, tracks3] : triple) {
+      LOGF(info, "Mixed event collisions: (%d, %d, %d)", c1.globalIndex(), c2.globalIndex(), c3.globalIndex());
+      count++;
+      if (count == 100)
+        break;
+
+      // Example of using tracks from mixed events -- iterate over all track pairs from the two collisions
```

Comment:
correct the comment :-), its three, not "two" collisions

Reply:
Hi @saganatt, this is a very nice update. I just had a couple of comments, only about adding an explanatory comments line basically. I think that for tutorials these cannot be too few. Thanks